pkgs/tools/typesetting/tex/mftrace/default.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, makeWrapper
+, autoreconfHook
+, buildEnv
+, python3
+, fontforge
+, potrace
+, texlive
+}:
+
+# to use with a texlive distribution, ensure that the desired fonts and
+# the packages kpathsea, t1utils, metafont are available at runtime
+#
+# mftrace can also be added to nixpkgs' texlive via texlive.combine
+#
+# accepted overrides:
+# - potrace = autotrace
+# - fontforge = ghostscript (limited functionality)
+# - fontforge = null (limited functionality)
+
+let self = stdenv.mkDerivation rec {
+  pname = "mftrace";
+  version = "1.2.20";
+
+  # https://lilypond.org/download/sources/mftrace/mftrace-1.2.20.tar.gz
+  # is incomplete, fetch repo and use autoconf instead
+  # see https://github.com/hanwen/mftrace/issues/13
+  src = fetchFromGitHub {
+    owner = "hanwen";
+    repo = "mftrace";
+    rev = "release/${version}";
+    sha256 = "02ik25aczkbi10jrjlnxby3fmixxrwm2k5r4fkfif3bjfym7nqbc";
+  };
+
+  nativeBuildInputs = [ makeWrapper autoreconfHook python3 potrace ];
+
+  buildInputs = [ fontforge potrace ];
+
+  postInstall = ''
+    wrapProgram $out/bin/mftrace --prefix PATH : ${lib.makeBinPath buildInputs}
+  '';
+
+  passthru.tlType = "bin";
```

Comment:
mftrace doesn't provide any texmf files, so I don't see why would you want to embed it into texlive. It should be perfectly functional as a standalone program.

Reply:
> It should be perfectly functional as a standalone program.

That's the thing, it doesn't work at all without t1utils and metafont. I am happy to drop this change, but, a last defence: I wrote this in order to let `texlive.combine` resolve mftrace's dependencies at installation time, instead of having the user figuring it out. The stronger approach would be to specify a custom combination in the `buildInputs`, but it cannot work due to kpathsea shenanigans. This is the next best thing as far as I can tell.

My underlying request is: the packages (mftrace, LaTeXML, possibly others?) that have a dependency on texlive, but cannot use `buildInputs`/`makeWrapper`, should be made able to pass their dependencies to `texlive.combine`. This orthogonal to providing texmf files, I think.

Reply:
I don't see what `texlive.combine` brings over the standard PATH combination mechanizm. This works fine:
```
nix-shell -I nixpkgs=`pwd`/nixpkgs -p mftrace -p texlive.combined.scheme-small --pure --run "mftrace cmr10"
```
But, of course, if you we can keep `tlType` if you want to run it another way.

Reply:
Thanks! I suppose I am to blame – I start from collection-basic, not scheme-small, and get dependency errors quite regularly.

Given that `combine` is not your standard `buildEnv`, something I only understood a few days ago, I will add a warning that `texlive.combine { mftrace ... }` is an experimental usage (at least until #118745 is fixed, and depending on how the fix looks like).